HughTheLightbulbV2/Device/mqtt.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright SCI Semiconductor and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#include "interface.hh"
+
+#include <NetAPI.h>
+#include <cstdlib>
+#include <debug.hh>
+#include <fail-simulator-on-error.h>
+#include <locks.hh>
+#include <mqtt.h>
+#include <platform-entropy.hh>
+#include <platform-gpio.hh>
+#include <platform-rgbctrl.hh>
+#include <sntp.h>
+#include <tick_macros.h>
+
+// Uncomment to use the demo.cheriot.org server instead of mosquitto.
+// This is faster (not rate limited) but might not be running.
+// #define CHERIOT_DEMO_SERVER 1
+
+#ifdef CHERIOT_DEMO_SERVER
+#	include "letsencrypt.h"
+#else
+#	include "mosquitto.org.h"
+#endif
+
+using CHERI::Capability;
+
+using Debug            = ConditionalDebug<true, "Hugh the lightbulb">;
+constexpr bool UseIPv6 = CHERIOT_RTOS_OPTION_IPv6;
+
+/// Maximum permitted MQTT client identifier length (from the MQTT
+/// specification)
+constexpr size_t MQTTMaximumClientLength = 23;
+/// Prefix for MQTT client identifier
+constexpr std::string_view clientIDPrefix{"cheriotMQTT"};
+/// Space for the random client ID.
+std::array<char, MQTTMaximumClientLength> clientID;
+static_assert(clientIDPrefix.size() < clientID.size());
+
+// MQTT network buffer sizes
+constexpr const size_t networkBufferSize    = 1024;
+constexpr const size_t incomingPublishCount = 20;
+constexpr const size_t outgoingPublishCount = 20;
+
+// MQTT test broker: https://test.mosquitto.org/
+// Note: port 8883 is encrypted and unautenticated
```

Comment:
```suggestion
// Note: port 8883 is encrypted and unauthenticated
```

Unauthenticated?  Isn't TLS getting us proof (up to X.509, anyway) of server identity?  Do you mean "accepts unauthenticated clients"?

Reply:
Does not authenticate clients.